### PR TITLE
Subscription

### PR DESCRIPTION
**Pull request by DarkerTH**
Created Subscription service.
Inserts into Watchlist on email submit.

### CODE REVIEW

File: src/AppBundle/Service/Subscription.php [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+    /**
+     * @return bool
+     */
+    public function validate(){
+
+        $validator = $this->getContainer()->get('validator');
+        $errors = $validator->validate($this);
+
+        if (count($errors) > 0) {
```
Comment (zyrukas) [inline src/AppBundle/Service/Subscription.php]: naudok empty() funkcija, sutaupysi performance, nes neskaiciuos visu objektu

**Comment by niumis** [comment]
+1

**Comment by DarkerTH** [comment]
@zyrukas @niumis Let's go boyz

File: src/AppBundle/Service/Subscription.php (hunk 1 above)
Comment (DarkerTH) [inline src/AppBundle/Service/Subscription.php]: Panaudojus empty() cia, gavosi taip, kad net jei count($errors) = 0, empty() returnina false. Tai pasilikau prie count()

**Comment by niumis** [comment]
Amazing!
+

File: src/AppBundle/Service/Subscription.php [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    /**
+     * @param int $brandId
+     * @return Subscription
+     */
+    public function setBrandId($brandId)
+    {
+        $this->brandId = $brandId;
+        return $this;
```
Comment (zyrukas) [inline src/AppBundle/Service/Subscription.php]: space before return

File: src/AppBundle/Service/Subscription.php [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    /**
+     * @param string $email
+     * @return Subscription
+     */
+    public function setEmail($email)
+    {
+        $this->email = $email;
+        return $this;
```
Comment (zyrukas) [inline src/AppBundle/Service/Subscription.php]: space

File: src/AppBundle/Service/Subscription.php [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    /**
+     * @param int $modelId
+     * @return Subscription
+     */
+    public function setModelId($modelId)
+    {
+        $this->modelId = $modelId;
+        return $this;
```
Comment (zyrukas) [inline src/AppBundle/Service/Subscription.php]: space

**Comment by zyrukas** [comment]
+1

**Pull request merged**